### PR TITLE
Mam disco barejid changes

### PR DESCRIPTION
**Pull request by JanuszJakubiec**
Adding test to check if sending disco to different jid results in errror.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/204720) / elasticsearch_and_cassandra_mnesia / d46cdecb2c9e5f66694275fa4f8c416f748d0a4f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4202/204720/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4202/204720/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40d5cd9f60303e.2024-01-03_08.44.34/index.html)
**OK: 380** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/204742) / elasticsearch_and_cassandra_mnesia / b9ca511f48c630b82ddecc61b33d0fddaf4ca58d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4202/204742/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4202/204742/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%4061ae2addee52.2024-01-03_08.46.55/index.html)
**OK: 380** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/204775) / elasticsearch_and_cassandra_mnesia / 3f93aa16a810d1ae0bf6ba71c88558d73909deef
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4202/204775/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4202/204775/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%401bd8ca7ab2ea.2024-01-03_08.52.37/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Review by chrzaszcz — APPROVED** [review]
Looks good, I added one comment

File: big_tests/tests/mam_SUITE.erl [hunk 1]
```diff
@@ -2947,6 +2951,16 @@ mam_service_discovery_to_client_bare_jid(Config) ->
         end,
     escalus_fresh:story(Config, [{alice, 1}], F).
 
+mam_service_discovery_to_different_client_bare_jid_results_in_error(Config) ->
+    _P = ?config(props, Config),
```
[the remark this answers is not in the document]
Comment (JanuszJakubiec) [reply big_tests/tests/mam_SUITE.erl:2955]: I'm not sure, but the line is present in other test functions, both above and below. I can try deleting it, but should I remove it from other test function if it is not needed?

[the remark this answers is not in the document]
Comment (chrzaszcz) [reply big_tests/tests/mam_SUITE.erl:2955]: I'd remove it, because it doesn't seem to do anything.

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/204833) / elasticsearch_and_cassandra_mnesia / c1828ec4ae1136a07f5034467e02cdf3ab382248
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4202/204833/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4202/204833/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40580347cb9652.2024-01-03_10.16.24/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Review by chrzaszcz — APPROVED** [review]
👌

**Pull request merged**